pkg/app/piped/executor/waitapproval/waitapproval.go

```diff
@@ -114,3 +118,40 @@ func (e *Executor) checkApproval(ctx context.Context) (string, bool) {
 	}
 	return approveCmd.Commander, true
 }
+
+func (e *Executor) reportDeploymentApproved(ctx context.Context) {
+	ds, err := e.TargetDSP.Get(ctx, e.LogPersister)
```

Comment:
Let's use `GetReadOnly` for better performance.

Reply:
6643184